cmd/describe/upgrade/cmd.go

```diff
@@ -38,7 +39,15 @@ var Cmd = &cobra.Command{
   rosa describe upgrade`,
 	Run:    run,
 	Hidden: false,
-	Args:   cobra.NoArgs,
+	Args: func(cmd *cobra.Command, argv []string) error {
```

Comment:
@Mischulee We are using the same function in multiple locations. Could you please refactor into a method that returns the function to be used. For example:

```
Args: NewUpgradeArgsFunction(),
```

where `NewUpgradeArgsFunction()` is a function that returns the `Args` func e.g:

```
func NewUpgradeArgsFunction() func(cmd *cobra.Command, argv []string) {
 return {
        flags := cmd.Flags()
	machinepoolID, _ := flags.GetString("machinepool")

	if !machinepool.MachinePoolKeyRE.MatchString(machinepoolID) {
		return fmt.Errorf("expected a valid identifier for the machine pool")
	}
	return nil
  }
}
```

Could you also the please add unit tests for the function to make sure it behaves as we expect. I'd suggest creating a new package in `pkg`, perhaps `upgrade` and moving the function and tests there.

There are lots of examples of how we use Ginkgo to write unit tests in the project. One that provides a good example of testing the flags on a `cobra.Command` is `pkg/output/flag_test.go`. Take a look and see if you are able to write your own test. If you have any other questions or need some pointers, please just connect with me on Slack.

cheers,

Rob

Reply:
@robpblake Thank you for a helpful review! I refactored the function and added unit test as suggested.